packages/react-native/React/Base/RCTUtils.m

```diff
@@ -565,7 +565,17 @@ BOOL RCTRunningInAppExtension(void)
   if (RCTRunningInAppExtension()) {
     return nil;
   }
-
+  
+  id<UIApplicationDelegate> delegate = RCTSharedApplication().delegate;
+  
+  SEL lastFocusedWindowSelector = NSSelectorFromString(@"lastFocusedWindow");
```

Comment:
This solution is not ideal but I didn't find anything better that's available publicly. 

I found private API that allows to retrieve "key" window scene (the last interacted with), but usage of private APIs is not allowed. 

I'm leaving this for reference:

```objc
SEL isKeyWindowSceneSelector = NSSelectorFromString(@"_isKeyWindowScene");
    if ([windowScene respondsToSelector:isKeyWindowSceneSelector]) {
      BOOL isKeyWindowScene = [windowScene performSelector:isKeyWindowSceneSelector];
      if (!isKeyWindowScene) {
        continue;
      }
    }
```

Reply:
Post on the Apple dev forums and see if you can get an official response, some folks have gotten private iOS APIs made public via discussions there.